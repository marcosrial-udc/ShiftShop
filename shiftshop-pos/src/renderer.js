import * as React from 'react';
import * as ReactDOM from 'react-dom';
import {Provider} from 'react-redux';
import {MuiThemeProvider} from '@material-ui/core';
import {SnackbarProvider} from 'notistack';
import {IntlProvider} from 'react-intl';

/* Async/await regenerator necesaries */
import 'core-js/stable';
import 'regenerator-runtime/runtime';

import './assets/scss/index.scss';
import theme from './theme'

import {App} from './modules/app';

import configureStore from './store';
import {initReactIntl} from "./i18n";
import './polyfills';

/* Configure store. */
const store = configureStore();

/* Configure i18n. */
const {locale, messages} = initReactIntl();

ReactDOM.render(
    <MuiThemeProvider theme={theme}>
      <Provider store={store}>
        <IntlProvider locale={locale} messages={messages}>
          <SnackbarProvider anchorOrigin={{vertical: 'top', horizontal: 'center'}}
                            autoHideDuration={2500}>
            <App/>
          </SnackbarProvider>
        </IntlProvider>
      </Provider>
    </MuiThemeProvider>,
    document.getElementById("root"));
