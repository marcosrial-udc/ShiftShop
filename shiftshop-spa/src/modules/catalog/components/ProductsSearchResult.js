import React from 'react';
import {useSelector} from 'react-redux';
import {FormattedMessage} from 'react-intl';
import PropTypes from 'prop-types';
import {Box, LinearProgress, Typography} from '@material-ui/core';

import {emptyBox, search} from '../../../assets/images';
import useStyles from '../styles/ProductsSearchResult';

import * as selectors from '../selectors';
import ProductList from './ProductList';

const ProductsSearchResult = ({searching, startSearch, stopSearch}) => {
    const classes = useStyles();

    const productsSearch = useSelector(selectors.getProductsSearch);

    if (searching) {
        return (
            <LinearProgress className={classes.spinner} />
        );
    }

    if (!productsSearch) {
        return (
            <div className={classes.emptyPlaceholder}>
                <Box className={classes.imageBox}>
                    <img className={classes.image} src={search} alt="No searchs yet"/>
                </Box>
                <Typography className={classes.emptyText} variant="body1">
                    <FormattedMessage id="project.catalog.ProductsSearchResult.emptySearch"/>
                </Typography>
            </div>
        );
    }

    if (productsSearch.result.items.length === 0) {
        return (
            <div className={classes.emptyPlaceholder}>
                <Box className={classes.imageBox}>
                    <img className={classes.image} src={emptyBox} alt="No products"/>
                </Box>
                <Typography className={classes.emptyText} variant="body1">
                    <FormattedMessage id="project.catalog.ProductsSearchResult.emptyProducts"/>
                </Typography>
            </div>
        );
    }

    return (
        <div className={classes.products}>
            <ProductList products={productsSearch.result} criteria={productsSearch.criteria}
                         startSearch={startSearch} stopSearch={stopSearch}/>
        </div>
    );
};

ProductsSearchResult.propTypes = {
    searching: PropTypes.bool.isRequired,
    startSearch: PropTypes.func.isRequired,
    stopSearch: PropTypes.func.isRequired
};

export default ProductsSearchResult;
