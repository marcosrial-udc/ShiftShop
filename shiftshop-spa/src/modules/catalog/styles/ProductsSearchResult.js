import {makeStyles} from '@material-ui/core';

const useStyles = makeStyles(theme => ({
    spinner: {marginTop: theme.spacing(3),},
    emptyPlaceholder: {
        padding: theme.spacing(1),
        textAlign: 'center'
    },
    imageBox:{padding: theme.spacing(3)},
    image: {
        maxWidth: '100%',
        [theme.breakpoints.down('sm')]: {maxHeight: '15rem'},
        [theme.breakpoints.up('md')]: {maxHeight: '20rem',},
    },
    emptyText: {fontStyle: 'italic'},
    products: {marginTop: theme.spacing(2)}
}));

export default useStyles;
