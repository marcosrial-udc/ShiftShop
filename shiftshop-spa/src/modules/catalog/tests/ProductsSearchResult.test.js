import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {IntlProvider} from 'react-intl';

import {initReactIntl} from "../../../i18n";

import ProductsSearchResult from '../components/ProductsSearchResult';

jest.mock('../components/ProductList', () => () => (<mockProductList/>));

const {messages} = initReactIntl();

describe('ProductsSearchResult snapshot', () => {

    const middlewares = [];
    const mockStore = configureStore(middlewares);

    const isSearching = {searching: true};
    const isNotSearching = {searching: false};


    const createProductsSearchResult = (store, props) => {

        return renderer.create(
            <Provider store={store}>
                <IntlProvider locale='en' messages={messages}>
                    <ProductsSearchResult {...props}/>
                </IntlProvider>
            </Provider>
        );

    };

    test('Searching products', () => {

        // Initialize mockstore with empty state
        const initialState = {catalog: {productsSearch: null}};
        const store = mockStore(initialState);

        const productsSearchResult = createProductsSearchResult(store, isSearching);

        expect(productsSearchResult.toJSON()).toMatchSnapshot();

    });

    test('Not initialized productsSearch', () => {

        // Initialize mockstore with empty state
        const initialState = {catalog: {productsSearch: null}};
        const store = mockStore(initialState);

        const productsSearchResult = createProductsSearchResult(store, isNotSearching);

        expect(productsSearchResult.toJSON()).toMatchSnapshot();

    });

    test('Empty items in productsSearch', () => {

        // Initialize mockstore with empty state
        const initialState = {catalog: {productsSearch: {result: {items: []}}}};
        const store = mockStore(initialState);

        const productsSearchResult = createProductsSearchResult(store, isNotSearching);

        expect(productsSearchResult.toJSON()).toMatchSnapshot();

    });

    test('Some items in productsSearch', () => {

        // Initialize mockstore with empty state
        const initialState = {catalog: {productsSearch: {result: {items: [{}, {}]}, criteria: {}}}};
        const store = mockStore(initialState);

        const productsSearchResult = createProductsSearchResult(store, isNotSearching);

        expect(productsSearchResult.toJSON()).toMatchSnapshot();

    });

});
