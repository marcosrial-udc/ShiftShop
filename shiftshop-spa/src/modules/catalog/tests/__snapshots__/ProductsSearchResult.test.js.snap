// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsSearchResult snapshot Empty items in productsSearch 1`] = `
<div
  className="makeStyles-emptyPlaceholder-2"
>
  <div
    className="MuiBox-root MuiBox-root-60 makeStyles-imageBox-3"
  >
    <img
      alt="No products"
      className="makeStyles-image-4"
      src="emptyBox.svg"
    />
  </div>
  <p
    className="MuiTypography-root makeStyles-emptyText-5 MuiTypography-body1"
  >
    No products were found
  </p>
</div>
`;

exports[`ProductsSearchResult snapshot Not initialized productsSearch 1`] = `
<div
  className="makeStyles-emptyPlaceholder-2"
>
  <div
    className="MuiBox-root MuiBox-root-29 makeStyles-imageBox-3"
  >
    <img
      alt="No searchs yet"
      className="makeStyles-image-4"
      src="search.svg"
    />
  </div>
  <p
    className="MuiTypography-root makeStyles-emptyText-5 MuiTypography-body1"
  >
    This is too empty, try to search something!
  </p>
</div>
`;

exports[`ProductsSearchResult snapshot Searching products 1`] = `
<div
  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-spinner-1 MuiLinearProgress-indeterminate"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
    style={Object {}}
  />
</div>
`;

exports[`ProductsSearchResult snapshot Some items in productsSearch 1`] = `
<div
  className="makeStyles-products-6"
>
  <mockProductList />
</div>
`;
